python/lsst/afw/geom/testUtils.py

```diff
@@ -91,3 +100,759 @@ def __iter__(self):
+        toEndpoint = forward.getToEndpoint()
+        frameSet = forward.getFrameSet()
+        invFrameSet = inverse.getFrameSet()
+
+        # properties
+        self.assertEqual(forward.getFromEndpoint(),
+                         inverse.getToEndpoint(), msg=msg)
+        self.assertEqual(forward.getToEndpoint(),
+                         inverse.getFromEndpoint(), msg=msg)
+        self.assertEqual(forward.hasForward(), inverse.hasInverse(), msg=msg)
+        self.assertEqual(forward.hasInverse(), inverse.hasForward(), msg=msg)
+
+        # transformations of one point
+        # we don't care about whether the transformation itself is correct
+        # (see checkTransformation), so inPoint/outPoint need not be related
+        rawInPoint = self.makeRawPointData(fromEndpoint.getNAxes())
+        inPoint = fromEndpoint.pointFromData(rawInPoint)
+        rawOutPoint = self.makeRawPointData(toEndpoint.getNAxes())
+        outPoint = toEndpoint.pointFromData(rawOutPoint)
+
+        # transformations of arrays of points
+        nPoints = 7  # arbitrary
+        rawInArray = self.makeRawArrayData(nPoints, fromEndpoint.getNAxes())
+        inArray = fromEndpoint.arrayFromData(rawInArray)
+        rawOutArray = self.makeRawArrayData(nPoints, toEndpoint.getNAxes())
+        outArray = toEndpoint.arrayFromData(rawOutArray)
+
+        if forward.hasForward():
+            self.assertEqual(forward.tranForward(inPoint),
+                             inverse.tranInverse(inPoint), msg=msg)
+            self.assertEqual(frameSet.tranForward(rawInPoint),
+                             invFrameSet.tranInverse(rawInPoint), msg=msg)
+            # Assertions must work with both lists and numpy arrays
+            assert_array_equal(forward.tranForward(inArray),
+                               inverse.tranInverse(inArray),
+                               err_msg=msg)
+            assert_array_equal(frameSet.tranForward(rawInArray),
+                               invFrameSet.tranInverse(rawInArray),
+                               err_msg=msg)
+
+        if forward.hasInverse():
+            self.assertEqual(forward.tranInverse(outPoint),
+                             inverse.tranForward(outPoint), msg=msg)
+            self.assertEqual(frameSet.tranInverse(rawOutPoint),
+                             invFrameSet.tranForward(rawOutPoint), msg=msg)
+            assert_array_equal(forward.tranInverse(outArray),
+                               inverse.tranForward(outArray),
+                               err_msg=msg)
+            assert_array_equal(frameSet.tranInverse(rawOutArray),
+                               invFrameSet.tranForward(rawOutArray),
+                               err_msg=msg)
+
+    def checkTransformFromMapping(self, fromName, toName):
+        """Check a Transform_<fromName>_<toName> using the Mapping constructor
+
+        Parameters
+        ----------
+        fromName, toName : string
+            one of self.endpointPrefixes
```

Comment:
So it's technically allowed (e.g., https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#sections) but in the Developer Guide I don't talk about it because, yeah, you can't go wrong spelling out _exactly_ what each parameter is.